src/modules/azure.rs

```diff
@@ -0,0 +1,278 @@
+
+    let json_file = File::open(&json_file_path).ok()?;
+    let mut reader = BufReader::new(json_file);
+    reader.read_to_end(&mut buffer).ok()?;
+
+    let bytes = buffer.as_mut_slice();
+    let decodedbuffer = bytes.strip_prefix(&[239, 187, 191]).unwrap_or(bytes);
+
+    if let Ok(parsed_json) = serde_json::from_slice(decodedbuffer) {
+        Some(parsed_json)
+    } else {
+        log::error!("Failed to parse json");
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::modules::azure::parse_json;
+    use crate::test::ModuleRenderer;
+    use ansi_term::Color;
+    use ini::Ini;
+    use std::fs::File;
+    use std::io::{self, Write};
+    use std::path::PathBuf;
+
+    use tempfile::TempDir;
+
+    fn generate_test_config(dir: &TempDir, azure_profile_contents: &str) -> io::Result<()> {
+        save_string_to_file(
+            dir,
+            azure_profile_contents.to_string(),
+            String::from("azureProfile.json"),
+        )?;
+
+        Ok(())
+    }
+    #[test]
+    fn subscription_set_correctly() -> io::Result<()> {
+        let dir = tempfile::tempdir()?;
+
+        let azure_profile_contents = r#"{
+            "installationId": "3deacd2a-b9db-77e1-aa42-23e2f8dfffc3",
+            "subscriptions": [
+              {
+                "id": "f3935dc9-92b5-9a93-da7b-42c325d86939",
+                "name": "Subscription 1",
+                "state": "Enabled",
+                "user": {
+                  "name": "user@domain.com",
+                  "type": "user"
+                },
+                "isDefault": true,
+                "tenantId": "f0273a19-7779-e40a-00a1-53b8331b3bb6",
+                "environmentName": "AzureCloud",
+                "homeTenantId": "f0273a19-7779-e40a-00a1-53b8331b3bb6",
+                "managedByTenants": []
+              },
+              {
+                "id": "f568c543-d12e-de0b-3d85-69843598b565",
+                "name": "Subscription 2",
+                "state": "Enabled",
+                "user": {
+                  "name": "user@domain.com",
+                  "type": "user"
+                },
+                "isDefault": false,
+                "tenantId": "0e8a15ec-b0f5-d355-7062-8ece54c59aee",
+                "environmentName": "AzureCloud",
+                "homeTenantId": "0e8a15ec-b0f5-d355-7062-8ece54c59aee",
+                "managedByTenants": []
+              },
+              {
+                "id": "d4442d26-ea6d-46c4-07cb-4f70b8ae5465",
+                "name": "Subscription 3",
+                "state": "Enabled",
+                "user": {
+                  "name": "user@domain.com",
+                  "type": "user"
+                },
+                "isDefault": false,
+                "tenantId": "a4e1bb4b-5330-2d50-339d-b9674d3a87bc",
+                "environmentName": "AzureCloud",
+                "homeTenantId": "a4e1bb4b-5330-2d50-339d-b9674d3a87bc",
+                "managedByTenants": []
+              }
```

Comment:
Please move these before the `"isDefault": true,` subscription and remove `isDefault` from one to test more branches.